src/main/java/org/folio/rest/impl/OaiPmhImpl.java

```diff
@@ -0,0 +1,146 @@
+    oai.getRequest().withIdentifier(id).withMetadataPrefix(metadataPrefix);
+
+    try {
+      String response = ResponseHelper.getInstance().writeToString(oai);
+      asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond422WithApplicationXml(response)));
+    } catch (Exception e) {
+      logger.error("Unexpected error happened while processing GetRecord verb request", e);
+      asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond500WithTextPlain(getErrorMessage())));
+    }
+  }
+
+  @Override
+  public void getOaiIdentifiers(String resumptionToken, String from, String until, String set, String metadataPrefix,
+                                Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+                                Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetOaiIdentifiersResponse.respond500WithTextPlain("The verb ListIdentifiers is not supported yet :(")));
+  }
+
+  @Override
+  public void getOaiMetadataFormats(String identifier, Map<String, String> okapiHeaders,
+                                    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetOaiMetadataFormatsResponse.respond500WithTextPlain("The verb ListMetadataFormats is not supported yet :(")));
+  }
+
+  @Override
+  public void getOaiSets(String resumptionToken, Map<String, String> okapiHeaders,
+                         Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    asyncResultHandler.handle(succeededFuture(GetOaiSetsResponse.respond500WithTextPlain("The verb ListSets is not supported yet :(")));
+  }
+
+  @Override
+  public void getOaiRepositoryInfo(Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+                                   Context vertxContext) {
+    try {
+      String repoName = System.getProperty(REPOSITORY_NAME);
+      String baseUrl = System.getProperty(REPOSITORY_BASE_URL);
+      String emails = System.getProperty(REPOSITORY_ADMIN_EMAILS);
+      if (repoName == null || baseUrl == null || emails == null) {
+        throw new IllegalStateException(String.format("One or more of the required repository configs missing: " +
+          "{repository.name: %s, repository.baseURL: %s, repository.adminEmails: %s}", repoName, baseUrl, emails));
+      }
+
+      OAIPMH oai = buildBaseResponse(VerbType.IDENTIFY)
+        .withIdentify(objectFactory.createIdentifyType()
+                                   .withRepositoryName(repoName + "_" + okapiHeaders.get(OKAPI_HEADER_TENANT))
+                                   .withBaseURL(baseUrl)
+                                   .withProtocolVersion(System.getProperty(REPOSITORY_PROTOCOL_VERSION, REPOSITORY_PROTOCOL_VERSION_2_0))
+                                   .withEarliestDatestamp(Instant.EPOCH.truncatedTo(ChronoUnit.SECONDS))
```

Comment:
Consider splitting this temporary logic out into a separate method, e.g. getEarliestDatestamp(), so that later we can swap out this for the real implementation which likely involves making calls to other modules.

Reply:
The same can be said for most of these fields... eventually the values we're pulling from system properties might be pulled from mod-configuration, etc.

Reply:
It make sense to do it, though the method signature will likely be changed later anyway, because we don't know yet what parameters will be required for the method to produce the value and declare it with no params for now.

Reply:
Done - the temporary logic was split into separate methods.